docs/RELEASE_ISSUE_TEMPLATE.md

```diff
@@ -14,7 +14,7 @@
 * Release reviewer: @who
 * Expected RC date: week of YYYY-MM-DD
 * 🚢 Expected final release date: YYYY-MM-DD
-* Accompanying PR for improving the release process: (example: https://github.com/ipfs/kubo/pull/9100)
```

Comment:
I am linking to this PR because we'll use it to set the right template in terms of PR title and description.